src/utils/event_loop.c

```diff
@@ -665,7 +670,7 @@ event_loop_run(struct event_loop *loop)
 
         if (loop->terminate_pending) {
             loop->terminate_pending = false;
-            break;
+            return loop->exit_code;
```

Comment:
What is the initial value of `loop->exit_code` when initialized? If nothing calls `event_loop_stop`, what value would this return? 0?

Reply:
That's a good point, if nothing calls event_loop_stop() then loop->exit_code is undefined. Currently loop->terminate_pending is only set by event_loop_stop() but that may change in the future. To prevent this being a bug, I'll open a PR that initializes exit_code to 0.

Reply:
#17